### PR TITLE
Removing old transactions from the queue

### LINKED ISSUE #3928: Transactions with "provided gas" > "block's max gas" keep propagating to other peers
Although I'm not quite familiar with Rust and cannot find the exact place which cause the issue, I'm collecting transaction broadcasts from the network and receiving them every 2-4 second from the same clients.

This tx: 0xf9c7e9165d9d55c060c4d098ff818bb182169227b21b59851d73ebcedfed0bee   provides 4.3 million gas, and Parity clients just sending it to each other.

Affected versions(at least):
"Parity/v1.4.5-beta-a028d04-20161126/x86_64-linux-gnu/rustc1.13.0"
"Parity/v1.4.6-beta-5feccf1-20161205/x86_64-linux-gnu/rustc1.13.0"

### PR DESCRIPTION
**Pull request by tomusdrw**
I'm not sure about the block time choice, or balance re-validation.
Would like to hear some second thoughts on this.

Closes #3928
Explains: #4007, #3996, #3358

### CODE REVIEW

File: ethcore/src/miner/transaction_queue.rs [hunk 1]
```diff
@@ -488,6 +495,11 @@ pub enum PrioritizationStrategy {
 	GasFactorAndGasPrice,
 }
 
+/// Point in time when transaction was inserted.
+/// (implying block numbers)
+pub type InsertionTime = u64;
```
Comment (gavofyork) [inline ethcore/src/miner/transaction_queue.rs]: If this is blocks, might want to change type to `BlockNumber`, and name to `QueuingPeriod`.

File: ethcore/src/miner/transaction_queue.rs [hunk 2]
```diff
@@ -794,16 +813,45 @@ impl TransactionQueue {
...
 			.chain(self.future.by_address.keys())
-			.cloned()
-			.collect::<HashSet<_>>();
+			.map(|sender| (*sender, fetch_account(sender)))
+			.collect::<HashMap<_, _>>();
 
-		for sender in senders {
-			self.remove_all(sender, fetch_nonce(&sender));
+		for (sender, details) in senders.iter() {
+			self.remove_all(*sender, details.nonce);
```
Comment (gavofyork) [inline ethcore/src/miner/transaction_queue.rs]: `remove_all` is a rather misleading name. there is very specific logic here that the name should imply. suggest `remove_dead` or even just `cull`.

**Comment by gavofyork** [comment]
minor grumbles.

**Comment by tomusdrw** [comment]
Grumbles addressed, renamed `InsertionTime` to `QueuingInstant`, cause it's actually used more as a point in time, not period.

**Pull request merged**